addons/web/static/tests/views/fields/date_field.test.js

```diff
@@ -0,0 +1,398 @@
+
+    expect(".o_datetime_picker").toHaveCount(0);
+    await contains(".o_field_date input").click();
+    await animationFrame();
+    expect(".o_datetime_picker").toHaveCount(1);
+
+    await fieldInput("char_field").click();
+    expect(".o_datetime_picker").toHaveCount(0);
+});
+
+test("toggle datepicker far in the future", async () => {
+    Partner._records[0].date = "9999-12-31";
+    await mountView({ type: "form", resModel: "res.partner", resId: 1 });
+
+    expect(".o_datetime_picker").toHaveCount(0);
+    await contains(".o_field_date input").click();
+    expect(".o_datetime_picker").toHaveCount(1);
+
+    // focus another field
+    await fieldInput("char_field").click();
+    expect(".o_datetime_picker").toHaveCount(0);
+});
+
+test("date field is empty if no date is set", async () => {
+    Partner._records[0].date = false;
+    await mountView({ type: "form", resModel: "res.partner", resId: 1 });
+
+    expect(".o_field_date input").toHaveCount(1);
+    expect(".o_field_date input").toHaveValue("");
+});
+
+test("set an invalid date when the field is already set", async () => {
+    await mountView({ type: "form", resModel: "res.partner", resId: 1 });
+
+    expect(".o_field_widget[name='date'] input").toHaveValue("02/03/2017");
+    await fieldInput("date").edit("invalid date");
+    expect(".o_field_widget[name='date'] input").toHaveValue("02/03/2017", {
+        message: "Should have been reset to the original value",
+    });
+});
+
+test("set an invalid date when the field is not set yet", async () => {
+    Partner._records[0].date = false;
+    await mountView({ type: "form", resModel: "res.partner", resId: 1 });
+
+    expect(".o_field_widget[name='date'] input").toHaveValue("");
+    await fieldInput("date").edit("invalid date");
+    expect(".o_field_widget[name='date'] input").toHaveValue("");
+});
+
+test("value should not set on first click", async () => {
+    Partner._records[0].date = false;
+    await mountView({ type: "form", resModel: "res.partner", resId: 1 });
+
+    await contains(".o_field_date input").click();
+    expect(".o_field_widget[name='date'] input").toHaveValue("");
+    await contains(getPickerCell(22)).click();
+
+    await contains(".o_field_date input").click();
+    expect(".o_date_item_cell.o_selected").toHaveText("22");
+});
+
+test("date field in form view (with positive time zone offset)", async () => {
+    mockTimeZone(2); // should be ignored by date fields
+    await mountView({ type: "form", resModel: "res.partner", resId: 1 });
+
+    onRpc("web_save", async (_, { args }) => {
+        expect.step(args[1].date);
+    });
+
+    expect(".o_field_date input").toHaveValue("02/03/2017");
+
+    // open datepicker and select another value
+    await contains(".o_field_date input").click();
+    expect(".o_datetime_picker").toHaveCount(1);
+    expect(".o_date_item_cell.o_selected").toHaveCount(1);
+
+    // select 22 Feb 2017
+    await zoomOut();
```

Comment:
also, since the helper is one line long, I'd just inline it everywhere it is used. This is just a suggestion, based on the paradigm that the less helpers the better